src/compiler/crystal/syntax/parser.cr

```diff
@@ -5662,11 +5662,14 @@ module Crystal
       check :","
 
       next_token_skip_space_or_newline
-      check :INSTANCE_VAR
+      check(%i(INSTANCE_VAR NUMBER))
```

Comment:
Please move this to a constant, otherwise it's allocating an array every time the parser reaches this.

Reply:
Actually, since you do a check on `@token.type` later on this not needed anymore.